namerd/iface/destination/src/main/scala/io/buoyant/namerd/iface/destination/DestinationService.scala

```diff
@@ -0,0 +1,150 @@
+import io.linkerd.proxy.destination.Destination
+import io.linkerd.proxy.destination.Update.OneofUpdate
+import io.linkerd.proxy.destination.Update.OneofUpdate.{NoEndpoints => EndpointsNone, _}
+import io.linkerd.proxy.destination._
+import io.linkerd.proxy.net.IPAddress.OneofIp
+import io.linkerd.proxy.net.{IPAddress, TcpAddress}
+import java.net.{InetAddress, InetSocketAddress}
+
+class DestinationService(
+  interpreter: NameInterpreter
+) extends Destination {
+  import DestinationService._
+
+  //Diff implementation for Address Set
+  class AddressDiffable extends Diffable[Set] {
+    override def diff[T](
+      left: Set[T],
+      right: Set[T]
+    ): Diff[Set, T] = AddressDiff(right -- left, left -- right)
+
+    override def empty[T]: Set[T] = Set.empty
+  }
+
+  case class AddressDiff[T](add: Set[T], remove: Set[T]) extends Diff[Set, T] {
+    override def patch(coll: Set[T]): Set[T] = coll ++ add -- remove
+
+    override def map[U](f: T => U): Diff[Set, U] = AddressDiff(add.map(f), remove.map(f))
+  }
+
+  implicit val _addressDiffable = new AddressDiffable
+
+  private[this] val log = Logger.get(getClass.getName)
+  override def get(req: GetDestination): Stream[Update] = {
+    log.info(s"stream initiated from request: $req")
+    if (!req.scheme.contains("k8s")) {
+      val res = Stream.mk[Update]
+      res.close(GrpcStatus.Unimplemented("Unknown scheme: " + req.scheme.getOrElse("unknown")))
+      return res
+    }
+
+    // This assumes that our dtab begins with the /svc prefix
+    val name = Path(Seq(Buf.Utf8("svc")) ++ req.path.map(Buf.Utf8(_)): _*)
+
+    val stream = Stream.mk[Update]
+    interpreter.bind(Dtab.empty, name).run.changes.respond {
+      case Activity.Pending => // Immediately send no endpoints when stream starts successfully
+        stream.send(mkNoEndpointsUpdate(false)); ()
+      case Activity.Failed(e) => // Send no endpoints message when we fail to get endpoints
+        log.info(s"address lookup failed: $e")
+        stream.send(mkNoEndpointsUpdate(false)); ()
+      case Activity.Ok(t) => t.eval match {
+        case None => // Service exists but cannot find addresses for service.
+          stream.send(mkNoEndpointsUpdate(true)); ()
+        case Some(value) => // Evaluate what diff of addresses and send add, remove and update requests.
+          val _ = updates(name, value).diff.respond {
+            case AddressDiff(add, remove) =>
+              if (add.nonEmpty) stream.send(mkAddUpdate(add)); ()
+              if (remove.nonEmpty) stream.send(mkRemoveUpdate(remove)); ()
+
+              /* If we don't have endpoints to add and remove that means we did not have addresses
+                 To begin with, therefore the service exists but we do not have address to update.
+                 Send a no endpoints message with exists set to true
+               */
+
+              if (add.isEmpty && remove.isEmpty) stream.send(mkNoEndpointsUpdate(true)); ()
+          }
+      }
+    }
+    stream
```

Comment:
I think overall the pipeline will look something like:

```
Activity[NameTree[Name.Bound]]
Activity[Option[Set[Name.Bound]] //eval
Activity[Var[Set[Addr]]] // Var.collect
Activity[Var[Addr.Bound]] // foldAddr
Var[Addr] // flatMap  (e.g. Activity.Pending -> Var(Addr.Pending))
Event[Addr] // changes
Event[Addr.Bound] // collect, dropping types other than Bound
Event[Set[Address]] // map
Event[Diff] // diff
Event[Update]  // eventFlatMap
```

One wrinkle that you'll run into is how to prepend a NoEndpoints(true) before the first Add in the event stream.  A custom combinator could do the trick:

```
def prependEvent[T](ev: Event[T], init: T): Event[T] = new Event[T] = {
  def register(w: Witness[T]): Closable = {
    w.notify(init)
    ev.respond(w.notify)
  }
}
```